Sources/RInAppMessaging/Repositories/CampaignRepository.swift

```diff
@@ -152,19 +146,14 @@ internal class CampaignRepository: CampaignRepositoryType {
 
     // MARK: - Helpers
 
-    private func findCampaign(withID id: String) -> Campaign? {
-        (allCampaigns.get() + tooltips.get()).first(where: { $0.id == id })
-    }
-
-    private func updateImpressionsLeftInCampaign(_ campaign: Campaign, newValue: Int) -> Campaign? {
+    private func updateImpressionsLeftInCampaign(withID id: String, by value: Int) -> Campaign? {
         var newList = allCampaigns.get()
```

Comment:
@imaginaris @Esakkiraja-Pothikannan  The newList should contain `allCampaigns.get() + tooltips.get()`?
`findCampaign(withID id: String)` is deleted and was searching the given campaign in `allCampaigns.get() + tooltips.get()`.

Reply:
It was discussed with @imaginaris. `findCampaign(withID id: String)` is the old implementation. Now `allCampaigns.get()` will return campaigns include `tooltips.get()`.